src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -341,72 +343,97 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             relativePath = eventPaths[i].Remove(0, _fullDirectory.Length);
                         }
 
-                        // Check if this is a rename
-                        if (IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
+                        var changedFlags = FSEventStreamEventFlags.kFSEventStreamEventFlagItemInodeMetaMod |
```

Comment:
the relativePath is used in all of the event notifications... I'll move the main event filtering logic to a helper so we can avoid this calculation until we know we have a valid event.